src/context/auth.context.tsx

```diff
@@ -0,0 +1,83 @@
+import { createContext, ReactNode, useEffect, useState } from "react";
+import { Customer } from "../interfaces/customers/customer.interface";
+import { useFetchCustomer } from "../hooks/customers/useFetchCustomer";
+
+export interface IAuthContext {
+  customer: Customer | null;
+  accessToken: string | null;
+  refreshToken: string | null;
+  login: (
+    customer: Customer,
+    accessToken: string,
+    refreshToken: string
+  ) => void;
+  logout: () => void;
+}
+
+interface AuthProviderProps {
+  children: ReactNode;
+}
+
+export const AuthContext = createContext<IAuthContext>({} as IAuthContext);
+
+export const AuthProvider = ({ children }: AuthProviderProps) => {
+  const [customer, setCustomer] = useState<Customer | null>(null);
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [refreshToken, setRefreshToken] = useState<string | null>(null);
+
+  const {
+    error,
+    customer: fetchedCustomer,
+    fetchCustomer,
+  } = useFetchCustomer();
+
+  const login = async (
+    customer: Customer,
+    accessToken: string,
+    refreshToken: string
+  ) => {
+    console.log(customer);
+
+    localStorage.setItem(
+      "user",
```

Comment:
Extract the 'user' in a constant for better maintainability. We have it in multiple places and if you want to change it you will then change it in one place.